client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -249,6 +285,98 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
         batchHandleChangePartitionRequestInterval,
         TimeUnit.MILLISECONDS)
     }
+
+    batchHandleCommitPartitionSchedulerThread.foreach {
+      _.scheduleAtFixedRate(
+        new Runnable {
+          override def run(): Unit = {
+            committedPartitionInfo.asScala.foreach { case (shuffleId, shuffleCommittedInfo) =>
+              batchHandleCommitPartitionExecutors.submit {
+                new Runnable {
+                  override def run(): Unit = {
+                    if (inProcessStageEndShuffleSet.contains(shuffleId) ||
+                      stageEndShuffleSet.contains(shuffleId)) {
+                      logWarning(s"Shuffle $shuffleId ended or during processing stage end.")
+                      shuffleCommittedInfo.commitPartitionRequests.clear()
```

Comment:
Should put this line inside shuffleCommittedInfo.synchronize

Reply:
Done